airflow_tecton/operators/tecton_feature_table_job_operator.py

```diff
@@ -0,0 +1,112 @@
+        templates_exts: Union[list[str], None] = None,
+        **kwargs,
+    ):
+        """
+
+        :param conn_id: Airflow connection ID for Tecton connection
+        :param workspace: Workspace of FeatureView
+        :param feature_view: FeatureView name
+        :param online: Whether job writes to online store
+        :param offline: Whether job writes to offline store
+        :param allow_overwrite: Whether jobs are able to run materialization
+               for periods that previously have materialized data. Note that
+               this can cause inconsistencies if the underlying data has changed.
+        :param df_generator: A reference to an object that is callable and
+               returns pandas.DataFrame
+        :param op_args: a list of positional arguments that will get unpacked when
+            calling df_generator
+        :param op_kwargs: a dictionary of keyword arguments that will get unpacked
+            in df_generator
+        :param templates_dict: a dictionary where the values are templates that
+            will get templated by the Airflow engine sometime between
+            ``__init__`` and ``execute`` takes place and are made available
+            in df generator's context after the template has been applied. (templated)
+        :param templates_exts: a list of file extensions to resolve while
+            processing templated fields, for examples ``['.sql', '.hql']``
+        :param kwargs: Airflow base kwargs passed to BaseOperator
+        """
+        super().__init__(**kwargs)
+        self.workspace = workspace
+        self.feature_view = feature_view
+        self.online = online
+        self.offline = offline
+        self.allow_overwrite = allow_overwrite
+        self.conn_id = conn_id
+        self.job_id = None
+        if df_generator and not callable(df_generator):
+            raise Exception("`df_generator` param must be callable")
+        self.df_generator = df_generator
+        self.op_args = op_args or ()
+        self.op_kwargs = op_kwargs or {}
+        self.templates_dict = templates_dict
+        if templates_exts:
+            self.template_ext = templates_exts
+
+    def execute(self, context) -> Any:
+        hook = TectonHook.create(self.conn_id)
+
+        if is_job_finished(hook, self.workspace, self.feature_view, self.online,
+                           self.offline, self.allow_overwrite, None, None, "ingest"):
+            return
+
+        resp = ingest_feature_table_with_pandas_df(hook, self.workspace, self.feature_view, context,
+                                                   self.df_generator, self.op_args, self.op_kwargs,
+                                                   self.templates_dict)
+
+        self.job_id = resp["job"]["id"]
+        check_job_status(hook, self.workspace, self.feature_view, self.job_id)
+
+    def on_kill(self) -> None:
+        kill_job(TectonHook.create(self.conn_id), self.workspace, self.feature_view, self.job_id)
```

Comment:
i think this needs ot check if job_id is set

Reply:
yeah, self_id is checked in kill_job function

Reply:
I think it's a little confusing like this. Better to only call kill_job if job_id is set